app/code/Magento/Customer/Controller/Adminhtml/Group/Save.php

```diff
@@ -99,7 +99,7 @@ public function execute()
                 } else {
                     $customerGroup = $this->groupDataFactory->create();
                 }
-                $customerGroup->setCode(!empty($customerGroupCode) ? $customerGroupCode : null);
+                $customerGroup->setCode(!empty($customerGroupCode) ? trim($customerGroupCode) : null);
```

Comment:
It's better to use `trim` on  the line where `customerGroupCode` retrieved from request